### PR TITLE
Bug 1925596: Text overflows Card boundary; Add Commit msg to details

### PR DESCRIPTION
**Pull request by keithchong**
Signed-off-by: Keith Chong <kykchong@redhat.com>

See GitOps 611. 

Screen Shot 2021-02-04 at 12 24 51 PM

### CODE REVIEW

**Comment by keithchong** [comment]
Solution.  Text will truncate and hover/tool tip will appear to show the entire string.

Screen Shot 2021-02-04 at 7 36 43 PM

**Comment by keithchong** [comment]
Fixes https://github.com/openshift/console/issues/8067

**Comment by divyanshiGupta** [comment]
@keithchong since we are adding truncation for image name and commit details in this PR it will be great if we can add truncation and tooltip for environment name as this will fix this issue https://issues.redhat.com/browse/ODC-4876 based on comment https://issues.redhat.com/browse/ODC-4876?focusedCommentId=15133883&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-15133883. Thanks!

cc: @lwrigh

**Comment by keithchong** [comment]
Hi @divyanshiGupta , I can work on the fix for that. However, I propose we create a separate PR for that so as to not hold up these changes, since they are independent.  Also, note the time constraints we have (I find testing will take up some time)

**Comment by divyanshiGupta** [comment]
@keithchong I am fine with creating another PR for adding truncation to env name.

@lwrigh what do you think?

**Comment by lwrigh** [comment]
The truncation in the labels looks good to me.

**Comment by keithchong** [comment]
@divyanshiGupta , I'm not sure why the test failed. Doesn't seem related to the changes.

**Comment by divyanshiGupta** [comment]
@keithchong it says this `i18n files are not up to date. Commit them to fix.`. Try running `yarn i18n` and then push the changes.

File: frontend/packages/gitops-plugin/src/components/details/CommitDetails.tsx [hunk 1]
```diff
@@ -80,11 +84,34 @@ const CommitDetails: React.FC<CommitDetailsProps> = ({ imageName }) => {
...
+            <SplitItem isFilled>
+              <Label className="odc-gitops-commit__item" isTruncated>
+                {t('gitops-plugin~{{id}}', { id: commitData.id })}
+              </Label>
+            </SplitItem>
+          </Split>
+          <Split className="odc-gitops-commit">
+            <SplitItem isFilled>
+              <Label className="odc-gitops-commit__item" isTruncated>
+                {t('gitops-plugin~{{msg}}', { msg: commitData.msg })}
```
Comment (jerolimov) [inline frontend/packages/gitops-plugin/src/components/details/CommitDetails.tsx:104]: If you only pass the msg it is not required to use i18n here. IMHO.

File: frontend/packages/gitops-plugin/src/components/details/CommitDetails.tsx [hunk 2]
```diff
@@ -80,11 +84,34 @@ const CommitDetails: React.FC<CommitDetailsProps> = ({ imageName }) => {
...
+            <SplitItem isFilled>
+              <Label className="odc-gitops-commit__item" isTruncated>
+                <Timestamp timestamp={commitData.timestamp} />
+              </Label>
+            </SplitItem>
+          </Split>
+          <Split className="odc-gitops-commit">
+            <SplitItem isFilled>
+              <Label className="odc-gitops-commit__item" isTruncated>
+                {t('gitops-plugin~{{id}}', { id: commitData.id })}
```
Comment (jerolimov) [inline frontend/packages/gitops-plugin/src/components/details/CommitDetails.tsx:97]: If you only pass the msg it is not required to use i18n here. IMHO.

**Comment by rohitkrai03** [comment]
/retitle Bug 1925596: Text overflows Card boundary; Add Commit msg to details

File: frontend/packages/gitops-plugin/locales/en/gitops-plugin.json [hunk 3]
```diff
@@ -1,5 +1,7 @@
 {
-  "{{id}} by {{author}}": "{{id}} by {{author}}",
+  "{{msg}}": "{{msg}}",
+  "{{id}}": "{{id}}",
```
Comment (jerolimov) [inline frontend/packages/gitops-plugin/locales/en/gitops-plugin.json:3]: I don't see a reason for this two attributes. Just skip i18n method in this case.

File: frontend/packages/gitops-plugin/src/components/details/CommitDetails.tsx (hunk 1 above)
Comment (keithchong) [reply frontend/packages/gitops-plugin/src/components/details/CommitDetails.tsx:104]: @jerolimov , I tinkered with adding `Message:` to this and `id:` to the id string, but they took up too much space. I just deleted them from here.  Thanks for catching this.

File: frontend/packages/gitops-plugin/locales/en/gitops-plugin.json (hunk 3 above)
Comment (keithchong) [reply frontend/packages/gitops-plugin/locales/en/gitops-plugin.json]: Removed this in the latest update.

File: frontend/packages/gitops-plugin/src/components/details/CommitDetails.tsx (hunk 2 above)
Comment (keithchong) [reply frontend/packages/gitops-plugin/src/components/details/CommitDetails.tsx]: This has been updated.

File: frontend/packages/gitops-plugin/src/components/details/CommitDetails.tsx (hunk 1 above)
Comment (keithchong) [reply frontend/packages/gitops-plugin/src/components/details/CommitDetails.tsx]: This has been updated.

**Review by rohitkrai03 — APPROVED** [review]
/lgtm
/approve

Tested locally. Works as expected.

**Pull request merged**